js/apps/admin-ui/src/util.ts

```diff
@@ -171,3 +171,18 @@ export const addTrailingSlash = (url: string) =>
   url.endsWith("/") ? url : url + "/";
 
 export const generateId = () => Math.floor(Math.random() * 1000);
+
+type PathParam = { [key: string]: string | null };
+
+export function generateUnencodedPath<Path extends string>(
+  originalPath: Path,
+  params: PathParam,
+): string {
+  const encodedParams: PathParam = {};
+  Object.entries(params).forEach(
+    ([k, v]) => (encodedParams[k] = encodeURIComponent(v || "")),
+  );
+
+  //@ts-ignore TODO: Fix type annotation once https://github.com/remix-run/react-router/pull/10719 is merged.
+  return generatePath(originalPath, encodedParams);
```

Comment:
We can remove the `@ts-ignore` here in favor of casting `encodedParams` to `any`. That should give us a little more type safety, as at least `originalPath` is still checked.
```suggestion
  // TODO: Fix type annotation once https://github.com/remix-run/react-router/pull/10719 is merged.
  return generatePath(originalPath, encodedParams as any);
```